### DISCUSSION TITLE
go back to previous page after login

### DISCUSSION THREAD

**Discussion by zahraHaghi** [vercel/next.js#11721] [Help]
I am using next js and react. when user  click on login link or  when user redirect to login page with router.push, I want to after lgoin ,user go back to previous page.how can I do that?

**Comment by Majdi-Alkhusha** [comment]
You can use `Router.back()` or if you want you can add to state history to check previous page

**Reply by scr2em** [reply]
Router.back() won't work as expected if the user opens the login page for the first time 
let's say I made a new tap and opened ***.com/login, going back will direct the user to the blank tap

**Reply by alex-r89** [reply]
Hi,

Did you have a solution for this?

I am also finding that if a user opens a new tab and navigates directly to my-url/login, `router.back()` will return them to the default chrome landing screen.

Ideally Id like logic along the lines of: `if back is the same domain, go back, else return to "/"`

**Reply by wildfrontend** [reply]
just check if ` document.referrer` matched `window.location.host`(this it is your domain) or redirect to your `home` page

```js
  if (window.history.length > 1 && document.referrer.indexOf(window.location.host) !== -1) {
      Router.back();
    } else {
      Router.replace(PATH.home);
    }
```
